mantidproject/mantid
# Add monitor X support for transmission monitors in the ISIS SANS GUI

~~**Do not merge or test before #20229 has been merged**~~
Fixes #20275

This PR addresses two issues:
1. Dynamic addition of mask files via the GUI
2. Being able to not only specify M3 or M4 but to be able to specify any monitor.

# To Test

#### Test  for "Dynamic addition of mask files via the GUI"

Please find test data here: smb://olympic/babylon5/Scratch/Anton/GUI_TEST
Add the data to the Mantid path

1. Open the ISIS SANS v2 experimental interface
1. Select `SANS2D` as the instrument (below the data table)
2. For the user file load: USER_SANS2D_154E_2p4_4m_M3_Xpress_8mm_SampleChanger_FRONT.txt
3. For the batch file load: batch_file.csv
4. Go the `Settings>Mask` tab
5. Ib the `Mask File Input` line edit field click Browse and browse to `MASK_SANS2D_REAR_Edges_16Mar2015.xml`. Add it
  * Confirm that it appears in the `Masking Information` table at the bottom
6. Go to the `Settings>State Diagnostic` tab and press `Update row selection
7.  Check `mask->mask_files` and confirm that the path to the mask file is at the end of the list


#### Test  for MX monitor for transmission

Please find test data here: smb://olympic/babylon5/Scratch/Anton/GUI_TEST
Add the data to the Mantid path

1. Open the ISIS SANS v2 experimental interface
1. Select `SANS2D` as the instrument (below the data table)
2. For the user file load: USER_SANS2D_154E_2p4_4m_M3_Xpress_8mm_SampleChanger_FRONT.txt
3. For the batch file load: batch_file.csv
4. Go the `Settings>Adjustment` tab
  * Confirm that `Transmission monitor` shows 3
5. Set it to 5
6. Go to the `Settings>State Diagnostic` tab and press `Update row selection
7.  Check `adjustment->calculate_transmission->transmission_monitor` and confirm that it has been set to 5. This does not make sense for SANS2D and will most likely lead to useless reductions, but this feature has been explicitly requested, especially for ZOOM and LARMOR


**Release Notes** 
No release notes required, since the GUI is being release in this cycle

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
